src/switchroot/ostree-mount-util.h

```diff
@@ -87,29 +87,26 @@ cleanup_free (void *p)
 static inline char *
 find_proc_cmdline_key (const char *cmdline, const char *key)
 {
-  char *ret = NULL;
-  size_t key_len = strlen (key);
-
-  const char *iter = cmdline;
-  while (iter != NULL)
+  const size_t key_len = strlen (key);
+  for (const char *iter = cmdline; iter;)
     {
       const char *next = strchr (iter, ' ');
-      const char *next_nonspc = next;
-      while (next_nonspc && *next_nonspc == ' ')
-        next_nonspc += 1;
       if (strncmp (iter, key, key_len) == 0 && iter[key_len] == '=')
         {
           const char *start = iter + key_len + 1;
           if (next)
-            ret = strndup (start, next - start);
-          else
-            ret = strdup (start);
-          break;
+            return strndup (start, next - start);
+
+          return strdup (start);
         }
-      iter = next_nonspc;
+
+      if (next)
+        next += strspn (next, " \t\n\r");
```

Comment:
I don't think we'll ever have newlines in the kernel cmdline right?  And carriage return aka `\r` is even weirder.  And above for the `strchr` we only handle spaces, so I'd argue we should be consistent and only use `strspn (next, " ")` here.

Reply:
But actually, why do we have a separate `strchr` and `strspn` here?  I think this just boils down to:

```
iter = strchr (iter, ' ');
if (iter)
  iter += strspn (iter, " ");
```
or so?

A bit like what we have in the other copy.

Reply:
Because the next pointer is used to identify the length of the karg in the strndup...

If the case where you are the final karg you can just use strdup, because you are at the end of the string...

Reply:
" \t\n\r" -> " " pushed